File: divide-conquer/searchMatrix.go
package divide_conquer

func searchMatrix(matrix [][]int, target int) bool {
	if matrix == nil {
		return false
	}
	r, c := len(matrix)-1, 0
	for r >= 0 && c >= 0 && r < len(matrix) && c < len(matrix[0]) {
		if matrix[r][c] == target {
			return true
		} else if matrix[r][c] > target {
			r--
		} else {
			c++
		}
	}
	return false
}

File: algo-java/src/dayplan/SearchMatrix.java
package dayplan;

/**
 * @author Zexho
 * @date 2021/9/24 9:21 下午
 */
public class SearchMatrix {
    public boolean searchMatrix(int[][] matrix, int target) {
        int r = matrix.length - 1, c = 0;
        while (r >= 0 && c < matrix[0].length) {
            if (matrix[r][c] > target) {
                r--;
            } else if (matrix[r][c] < target) {
                c++;
            } else {
                return true;
            }
        }
        return false;
    }
}
